client/src/Containers/QuizArea.jsx

```diff
@@ -1,17 +1,214 @@
+  const [isLoading, setIsLoading] = useState(true);
+  const [index, setIndex] = useState(-1);
+  const [time, setTime] = useState(0);
+  const [timeGivenforQ, setTimeGivenforQ] = useState(0);
+  const [quizData, setQuizData] = useState([]);
+  const [selectedButton, setSelectedButton] = useState('');
+  const [showAnswer, setShowAnswer] = useState(false);
+  const [score, setScore] = useState(0);
+
+  const timeRef = useRef(-1);
+  const selectdButtonRef = useRef(null);
+  const correctOptionRef = useRef(null);
+  const showAnswerRef = useRef(false);
+
+  let interval;
+
+
+
+  // This function is used to color the option button when it is selected
+  function colorButton(optionID , bgcolor , color)
+  {
+      if(optionID === '') // if option id is empty string then selected option would be undefined
+      return; // So, we are exiting from function before any error
+
+
+      const slctdOption = document.getElementById(optionID); // Get the button with given option id
+      slctdOption.style.backgroundColor = bgcolor; // change the background color
+      slctdOption.style.color = color; // change the font color
+  }
+
+
+
+
+  // This function is used to color the button based on circumstances
+  function changeButton(optionID)
+  {
+      if( showAnswer )
+      return;
+
+    // If the selected button is selected again
+      if(selectedButton === optionID)
+      {
+        colorButton(selectedButton , 'white' , 'black'); // uncolor the button
+        setSelectedButton(''); // and set selected button to null
+        selectdButtonRef.current = '';
+    }
+    // If any other button is selected again
+    else
+    {
+        colorButton(optionID , '#2196f3' , 'white'); // color the currently selected button
+        colorButton(selectedButton , 'white' , 'black'); // uncolor the previously selected button
+        setSelectedButton(optionID); // set the currently selected button
+        selectdButtonRef.current = optionID;
+      }
+  }
+
+  function checkAnswer()
+  {
+    
+    clearInterval(interval); // [ NOT WORKING ] stop the timer
+
+    showAnswerRef.current = true;
+    setShowAnswer(showAnswerRef.current); // set the state of page to show the answer
+
+    if(selectdButtonRef.current === correctOptionRef.current) // if the option selected by user is correct option
+    {
+        colorButton(correctOptionRef.current , '#00e676' , 'white'); // show the correct option in green
+        setScore( prevscore => {
+            let newscore = timeRef.current / timeGivenforQ * 1000;
+            newscore += prevscore;
+            return newscore;
+        }); // update the score
+    }
+    else // if the option selected by user is incorrect
+    {
+        colorButton(correctOptionRef.current , '#00e676' , 'white'); // show the correct option in green
+        colorButton(selectdButtonRef.current , '#f50057' , 'white'); // show the selected incorrect option in red
+    }
+  }
+
+  
+  useEffect(() => {

```

Comment:
Above each useEffect make sure to add comments mentioning the purpose of useEffect.